glean-core/src/lib.rs

```diff
@@ -251,16 +351,137 @@ impl Glean {
 #[cfg(test)]
 mod test {
     use super::*;
+    use crate::metrics::StringMetric;
+
+    const GLOBAL_APPLICATION_ID: &str = "org.mozilla.glean.test.app";
+    pub fn new_glean() -> (Glean, tempfile::TempDir) {
+        let dir = tempfile::tempdir().unwrap();
+        let tmpname = dir.path().display().to_string();
+        let glean = Glean::new(&tmpname, GLOBAL_APPLICATION_ID, true).unwrap();
+        (glean, dir)
+    }
 
     #[test]
     fn path_is_constructed_from_data() {
-        let t = tempfile::tempdir().unwrap();
-        let name = t.path().display().to_string();
-        let glean = Glean::new(&name, "org.mozilla.glean", true).unwrap();
+        let (glean, _) = new_glean();
 
         assert_eq!(
-            "/submit/org-mozilla-glean/baseline/1/this-is-a-docid",
+            "/submit/org-mozilla-glean-test-app/baseline/1/this-is-a-docid",
             glean.make_path("baseline", "this-is-a-docid")
         );
     }
+
```

Comment:
Should we have a test ensuring that nothing changes if `set_upload_enabled` is called twice with the same value (`true, true` or `false, false`)?

Reply:
Yes -- glean-ac had this test but it involved mocking to make sure specific methods weren't called.  I'll have to figure out how to do that in Rust.

Reply:
Mocking in Rust is (for obvious reasons) awfully difficult and invasive and seemed like overkill for this.  So I just ended up adding a return value to `set_upload_enabled`...  Not ideal, but it works...

Reply:
Should we have a bug (if we don't have one already) about investigating mocking options for Rust?